tools/merge-volume-into-data-layer/main.py

```diff
@@ -0,0 +1,129 @@
+    tracing_tmpdir_path = extract_tracing_zip(args)
+
+    tracing_dataset = wkw.Dataset.open(os.path.join(tracing_tmpdir_path, '1'))
+    segmentation_dataset = wkw.Dataset.open(os.path.join(args.segmentation_path))
+
+    assert(tracing_dataset.header.num_channels == segmentation_dataset.header.num_channels)
+
+    tracing_bboxes = file_bboxes(tracing_dataset)
+    segmentation_bboxes = file_bboxes(segmentation_dataset)
+
+    bboxes_grouped = group_tracing_bboxes(tracing_bboxes, segmentation_bboxes)
+
+    print("Found {} tracing files, which will affect {} segmentation files".format(len(tracing_bboxes), len(bboxes_grouped)))
+
+    segmentation_file_len_voxels = segmentation_dataset.header.block_len * segmentation_dataset.header.file_len
+
+    count = 1
+    for bbox_group_key in bboxes_grouped:
+        segmentation_bbox = bboxes_grouped[bbox_group_key][0]
+        tracing_bboxes = bboxes_grouped[bbox_group_key][1]
+
+        print("Reading segmentation file {} of {}, bounding box: {}...".format(count, len(bboxes_grouped), segmentation_bbox))
+        data = segmentation_dataset.read(segmentation_bbox[0], segmentation_bbox[1])
+
+        print("  Overwriting tracing buckets in memory...")
+        for tracing_bbox in tracing_bboxes:
+            print("    Overwriting", tracing_bbox)
+            tracing_data = tracing_dataset.read(tracing_bbox[0], tracing_bbox[1])
+            topleft = list(map(lambda x: x % segmentation_file_len_voxels, tracing_bbox[0]))
+            shape = tracing_bbox[1]
+            bottomright = list( map(add, topleft, shape) )
+            # print("      Broatcasting to 0:1 {}:{}, {}:{}, {}:{}".format(topleft[0], bottomright[0], topleft[1], bottomright[1], topleft[2], bottomright[2]))
+            data[0:1, topleft[0]:bottomright[0], topleft[1]:bottomright[1], topleft[2]:bottomright[2]] = tracing_data
+
+        print("  Writing segmentation file back to disk...")
+        segmentation_dataset.write([0, 0, 0], data)
+        count = count + 1
+    print("Done")
+
+def confirmation_prompt(args):
+    if not args.yes:
+        answer = input("Are you sure you want to modify the data in {}? This cannot be undone. To continue, type “yes”: ".format(args.segmentation_path))
+        if answer != "yes":
+            print("Aborting")
+            sys.exit(0)
+
+def extract_tracing_zip(args):
+    tracing_tmpdir_path = tmp_filename()
+    os.makedirs(tracing_tmpdir_path)
+    with zipfile.ZipFile(args.tracing_path) as outer_zip:
+        if 'data.zip' in outer_zip.namelist():
+            outfile_path = os.path.join(tracing_tmpdir_path, 'data.zip')
+            with outer_zip.open('data.zip') as data_zip, open(outfile_path, 'wb') as outfile:
+                shutil.copyfileobj(data_zip, outfile)
+                extract_data_zip(outfile_path, tracing_tmpdir_path)
+            os.remove(outfile_path)
+        else:
+            extract_data_zip(args.tracing_path)
+
+    tracing_tmpdir_path = 'tmp-67X8KZUFP0'
```

Comment:
generate randomly? this might otherwise cause problems when merging multiple tracings at the same time

Reply:
ok, i see you have `tmp_filename` but don't use it here ;)